hub/api/tests/test_api.py

```diff
@@ -178,17 +178,43 @@ def test_empty_samples(ds: Dataset):
 
 @parametrize_all_dataset_storages
 def test_scalar_samples(ds: Dataset):
-    tensor = ds.create_tensor("scalars")
+    tensor = ds.create_tensor("scalars", dtype=np.uint8)
+    assert tensor.meta.dtype == "uint8"
 
-    tensor.append(5)
-    tensor.append(10)
-    tensor.append(-99)
+    # safe cast
+    tensor.append(255)
```

Comment:
Any way to assert that this won't warn? I feel like this _should_ be interpreted as an `np.int64` and then downcast with a warning. Any reason this doesn't occur? Is `tensor.append(256)` safe? If not, where are we checking the value to tell the difference (or does numpy handle it)?